### PR TITLE
More clear error message on uncompressing block

### PR DESCRIPTION
**Pull request by rickif**
The origin error message of uncompressing block is confusing, which may result from either build support or data corruption.

### CODE REVIEW

File: table/format.cc [hunk 1]
```diff
@@ -360,8 +360,13 @@ Status UncompressBlockContentsForCompressionType(
       UncompressData(uncompression_info, data, n, &uncompressed_size,
                      GetCompressFormatForVersion(format_version), allocator);
   if (!ubuf) {
+    if (!CompressionTypeSupported(uncompression_info.type()))
+      return Status::NotSupported(
+          "Unsupported compression method for this build",
+          CompressionTypeToString(uncompression_info.type()));
+
     return Status::Corruption(
```
Comment (mrambacher) [inline table/format.cc:368]: For code clarity, this might be better as an if-else

Comment (rickif) [reply table/format.cc]: That's a better style. I've updated the PR.

**Review by pdillinger — APPROVED** [review]
LGTM, Thanks!

File: table/format.cc (hunk 1 above)
Comment (rickif) [reply table/format.cc]: @mrambacher PTAL!

**Review by mrambacher — APPROVED** [review]
LGTM.  Thanks for the changes!

**Pull request closed without merging**